builtin/providers/google/resource_dataflow_job_test.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataflowJobCreate(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDataflowJob,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(
+						"google_dataflow_job.big_data"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDataflowJobDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_dataflow_job" {
+			continue
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		job, _ := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if job != nil {
+			return fmt.Errorf("Job still present")
+		}
+	}
+
+	return nil
+}
+
+func testAccDataflowJobExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Job does not exist")
+		}
+
+		return nil
+	}
+}
+
+var testAccDataflowJob = fmt.Sprintf(`
```

Comment:
Ok, let me check my understanding. Are these facts correct?
* A dataflow job has both a name and an id.
* The id is generated by the server and the name is user-specified
* Once a job is created, it never fully goes away- it just stops (either because it was finished, it was canceled, or it was drained)
* A job can be run multiple times, but only one instance of a job can be run at a time

If all that is true, I'm still not sure what the difference is between using the name vs the id as the terraform id. I think what you could do is remove the job from state if it's finished, canceled, or drained (by doing this in the read function) and then that would allow you to create it anew. You could also choose not to do that in the read function and only do that in delete, and then you would have to explicitly remove it from your config (or terraform destroy) each time you want to re-launch it, but that would also prevent against accidentally relaunching it once something changes.

Sorry to keep nagging about details, but I really want to make sure we get this right. The more information you can supply about your expected workflow the better. I'm looking for things like:
* Do you expect to run each job once? On a schedule? On an as-needed basis?
* Do you keep your dataflow configs together with the rest of your terraform configs? If so, what do you expect to happen when you run terraform plan/apply to update a different element?
* What are you expecting to need to do in order to relaunch a job? Should you have to explicitly remove it from your config (or run terraform destroy), or should running apply on a config where the job has ended be sufficient?

I promise I won't be upset if you give me too much detail :)